app/src/main/report/components/ReportsList/ReportsList.ts

```diff
@@ -0,0 +1,50 @@
+import { connect } from 'react-redux';
+import { Dispatch } from "redux";
+import { compose, lifecycle, branch, renderComponent } from "recompose";
+import { clone } from "lodash";
+
+import {ReportAppState} from "../../reducers/reportAppReducers";
+import { reportsActions } from "../../actions/reportsActions";
+import {ReportsListComponent, ReportsListComponentProps} from "./ReportsListComponent";
+import {Report} from "../../../shared/models/Generated";
+import {ReportsSortingFields} from "../../actionTypes/ReportsActionsTypes";
+import { LoadingElement } from "../../../shared/partials/LoadingElement/LoadingElement";
+
+export interface ReportsListContainerProps extends ReportsListComponentProps {
+    getReports: () => void;
+    ready: boolean;
+}
+
+export const getDisplayedReportsList = (items: Report[], sortBy: ReportsSortingFields) => {
+    if (!items || !items.length) return [];
+    let displayItemsSorted = clone(items).sort(sortReports(sortBy));
+    return displayItemsSorted;
+}
+
+export const sortReports = (sortBy: ReportsSortingFields) => (a: Report, b: Report)=> (a[sortBy] < b[sortBy] ? -1 : 1);
```

Comment:
See comment on the tests - this logic will have to vary depending on whether we're sorting by name or by date